From d0f2377ce2accd611610996c20810c074af51dda Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:30:21 -0400
Subject: [net] openvswitch: Fix memory leak in ovs_vport_alloc() error path

Message-id: <44d739cfbda8db05ec898a5a74319d3a641087ee.1410531783.git.jbenc@redhat.com>
Patchwork-id: 94009
O-Subject: [RHEL7.1 net PATCH 48/50] openvswitch: Fix memory leak in ovs_vport_alloc() error path
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 3791b3f6fb74d265c93d493d9bbf29c1e769ceae
Author: Christoph Jaeger <cj@linux.com>
Date:   Tue Aug 12 09:27:57 2014 +0200

    openvswitch: Fix memory leak in ovs_vport_alloc() error path

    ovs_vport_alloc() bails out without freeing the memory 'vport' points to.

    Picked up by Coverity - CID 1230503.

    Fixes: 5cd667b0a4 ("openvswitch: Allow each vport to have an array of 'port_id's.")
    Signed-off-by: Christoph Jaeger <cj@linux.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/vport.c b/net/openvswitch/vport.c
index 2b8e478..6e4f9d2 100644
--- a/net/openvswitch/vport.c
+++ b/net/openvswitch/vport.c
@@ -137,8 +137,10 @@ struct vport *ovs_vport_alloc(int priv_size, const struct vport_ops *ops,
  vport->ops = ops;
  INIT_HLIST_NODE(&vport->dp_hash_node);
 
- if (ovs_vport_set_upcall_portids(vport, parms->upcall_portids))
+ if (ovs_vport_set_upcall_portids(vport, parms->upcall_portids)) {
+  kfree(vport);
   return ERR_PTR(-EINVAL);
+ }
 
  vport->percpu_stats = alloc_percpu(struct pcpu_tstats);
  if (!vport->percpu_stats) {
-- 
1.7.1